hardware_interface/test/fake_components/test_generic_system.cpp

```diff
@@ -1027,3 +1065,83 @@ TEST_F(TestGenericSystem, generic_system_2dof_functionality_with_offset_custom_i
   EXPECT_EQ(
     states_map["GenericSystem2dof"].label(), hardware_interface::lifecycle_state_names::INACTIVE);
 }
+
+TEST_F(TestGenericSystem, valid_urdf_ros2_control_system_robot_with_gpio_)
+{
+  auto urdf = ros2_control_test_assets::urdf_head +
+              valid_urdf_ros2_control_system_robot_with_gpio_ + ros2_control_test_assets::urdf_tail;
+  hardware_interface::ResourceManager rm(urdf);
+
+  // check is hardware is configured
+  std::unordered_map<std::string, rclcpp_lifecycle::State> states_map;
+  states_map = rm.get_components_states();
+  EXPECT_EQ(
+    states_map["GenericSystem2dof"].label(), hardware_interface::lifecycle_state_names::INACTIVE);
+
+  // Check initial values
+  hardware_interface::LoanedStateInterface gp1_i1_s =
+    rm.claim_state_interface("flange_analog_IOs/analog_output1");
+  hardware_interface::LoanedStateInterface gp1_i2_s =
+    rm.claim_state_interface("flange_analog_IOs/analog_input1");
+  hardware_interface::LoanedStateInterface gp1_i3_s =
+    rm.claim_state_interface("flange_analog_IOs/analog_input2");
+  hardware_interface::LoanedStateInterface gp2_i1_s =
+    rm.claim_state_interface("flange_vacuum/vacuum");
+  hardware_interface::LoanedCommandInterface gp1_i1_c =
+    rm.claim_command_interface("flange_analog_IOs/analog_output1");
+  hardware_interface::LoanedCommandInterface gp2_i1_c =
+    rm.claim_command_interface("flange_vacuum/vacuum");
```

Comment:
Variable names do not make much sense, maybe something like:

```suggestion
  hardware_interface::LoanedStateInterface gpio1_a_o1_s =
    rm.claim_state_interface("flange_analog_IOs/analog_output1");
  hardware_interface::LoanedStateInterface gpio1_a_i1_s =
    rm.claim_state_interface("flange_analog_IOs/analog_input1");
  hardware_interface::LoanedStateInterface gpio1_a_o2_s =
    rm.claim_state_interface("flange_analog_IOs/analog_input2");
  hardware_interface::LoanedStateInterface gpio2_vac_s =
    rm.claim_state_interface("flange_vacuum/vacuum");
  hardware_interface::LoanedCommandInterface gpio1_a_o1_c =
    rm.claim_command_interface("flange_analog_IOs/analog_output1");
  hardware_interface::LoanedCommandInterface gpio2_vac_c =
    rm.claim_command_interface("flange_vacuum/vacuum");
```